### DISCUSSION TITLE
Developing a new Extension: Dev Documentation and Code Samples for Open Cart 4

### DISCUSSION THREAD

**Discussion by neeooh** [opencart/opencart#12324] [Q&A]
Hello everyone!

I would like to develop a new (simple) extension for OpenCart 4.

Can you please point me to developer documentation and some code samples?

I can't find anything on the official website.. https://docs.opencart.com/developer/module/

It would be nice to see how to convert XMLs (plenty of already developed extensions that don't work anymore on OC 4) to the newly introduced event system.

Thanks in advance!

**Comment by SDG1212** [comment]
Hello!

You can watch the repository with examples. It's a great start point how you can start to understand how to develop modules & themes and after that extensions.

https://github.com/opencart/examples

All in all, the only effective way to understand CMS is to analyze engine files. I can't call the new event system perfect, but it works. Maybe after some time we will see all the benefits of this system but now it needs more improvements and fixes.

**Comment by neeooh** [comment]
Thanks, however this seems to be for the previous version 3, not the newest one v4?

**Reply by SDG1212** [reply]
The link which you can see in my previous answer is for 4 version.
